src/components/Choropleth.vue

```diff
@@ -0,0 +1,152 @@
+import { storeToRefs } from "pinia";
+import { GeoJSON, Layer } from "leaflet";
+import { LGeoJson, LMap, LTileLayer } from "@vue-leaflet/vue-leaflet";
+import { Feature } from "geojson";
+import { useAppStore } from "../stores/appStore";
+import { useColourScale } from "../composables/useColourScale";
+import "leaflet/dist/leaflet.css";
+
+interface FeatureWithColour {
+    feature: Feature;
+    colour: string;
+}
+
+const FEATURE_ID_PROP = "shapeID";
+const FEATURE_NAME_PROP = "shapeName";
+const FEATURE_COUNTRY_PROP = "shapeGroup";
+
+const style = {
+    className: "geojson"
+};
+
+const backgroundLayer = {
+    url: "https://server.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Base/MapServer/tile/{z}/{y}/{x}",
+    attribution: "Tiles &copy; Esri &mdash; Esri, DeLorme, NAVTEQ",
+    maxZoom: 10,
+    minZoom: 3
+};
+
+const map = ref<typeof LMap | null>(null);
+const featureRefs = ref<(typeof LGeoJson)[]>([]);
+
+const { selectedFeatures, selectedIndicators, loading, selectedIndicator } = storeToRefs(useAppStore());
+const { selectCountry } = useAppStore();
+
+const { colourScales, getColour } = useColourScale(selectedIndicators);
+
+const featureId = (feature: Feature) => feature.properties![FEATURE_ID_PROP];
+const featureName = (feature: Feature) => feature.properties![FEATURE_NAME_PROP];
+
+const getColourForFeature = (feature, indicator) => {
+    const featureIndicators = selectedIndicators.value[featureId(feature)];
+    return getColour(indicator, featureIndicators);
+};
+
+// TODO: This structure is an inefficient way to get reactivity working, but we can probably do better!
+const featuresWithColours = computed(() => {
+    if (loading.value) {
+        return [];
+    }
+    const selectedInd = selectedIndicator.value;
+    return selectedFeatures.value.map((feature) => {
+        return {
+            feature,
+            colour: getColourForFeature(feature, selectedInd)
+        };
+    });
+});
+
+const updateBounds = () => {
+    if (!loading.value) {
+        if (map.value && map.value.leafletObject) {
+            map.value.leafletObject.fitBounds(
+                featuresWithColours.value.map((f: FeatureWithColour) => new GeoJSON(f.feature).getBounds())
+            );
+        }
+    }
+};
+
+const initialising = computed(() => {
+    return (
+        loading.value &&
+        !featuresWithColours.value.length &&
+        !selectedIndicators.value.length &&
+        !Object.keys(colourScales.value).length
+    );
+});
+
+// TODO: pull out tooltips stuff into composable when fully implement
+const tooltipForFeature = (feature: Feature) => {
+    let indicatorValues = "";
+    const fid = featureId(feature);
+    if (fid in selectedIndicators.value) {
+        const featureValues = selectedIndicators.value[fid];
+        indicatorValues = Object.keys(featureValues)
+            .map((key) => {
+                return `${key}: ${featureValues[key].mean} (+/- ${featureValues[key].sd})<br/>`;
```

Comment:
Round the numbers here to 2dp or 2sf perhaps?

Possibly (later?) indicate which is the indicator visible by styling it differently?